LDAR_Sim/model_code/methods/base_crew.py

```diff
@@ -0,0 +1,402 @@
+                    x_temp = np.float(site['lon'])
+                    y_temp = np.float(site['lat'])
+                    x_cut = self.state['current_x']
+                    y_cut = self.state['current_y']
+                    home_loc, distance = find_homebase_opt(
+                        x_temp, y_temp, x_cut, y_cut, self.HX, self.HY)
+
+                    self.state['current_x'] = home_loc[0]
+                    self.state['current_y'] = home_loc[1]
+                    minutes_remaining = (
+                        projected_end_time - self.allowed_end_time).total_seconds()/60
+                    self.rollover = []
+                    self.rollover.append(site)
+                    self.rollover.append(minutes_remaining)
+                    self.state['t'].current_date = self.allowed_end_time
+                    break
+                else:
+                    crew_x = np.float(site['lon'])
+                    crew_y = np.float(site['lat'])
+                    self.visit_site(site, travel_time)
+                self.worked_today = True
+        # ----------------------------------------------------------
+        else:
+            # self.state['t'].current_date += timedelta(
+            #     minutes=int((np.random.choice(self.config['t_bw_sites']))))
+            while self.state['t'].current_date < self.allowed_end_time:
+                facility_ID, found_site, site, travel_time = self.choose_site(0, 0)
+                if not found_site:
+                    break  # Break out if no site can be found
+
+                # Check to make sure there's enough time left in the day to do this site
+                # This projection includes the time it would time to travel back to the home base
+                if found_site:
+                    projected_end_time = self.state['t'].current_date + \
+                        timedelta(minutes=int(site['{}_time'.format(m_name)]))
+                    travel_home = timedelta(minutes=int(
+                        np.random.choice(self.config['t_bw_sites'])))
+                    if (projected_end_time + travel_home) > self.allowed_end_time:
+                        # There's not enough time left for that site today
+                        # - get started and figure out how much time remains
+                        minutes_remaining = (
+                            projected_end_time - self.allowed_end_time).total_seconds()/60
+                        self.rollover = []
+                        self.rollover.append(site)
+                        self.rollover.append(minutes_remaining)
+                        self.state['t'].current_date = self.allowed_end_time
+
+                    # There's enough time left in the day for this site
+                    elif (projected_end_time + travel_home) <= self.allowed_end_time:
+                        self.visit_site(site, travel_time)
+                    self.worked_today = True
```

Comment:
I'm particularly worried about how complex this code is... @AkaMrmuffin  have you extensively tested this out with many different scenarios to see if the results make sense? In particular, what happens if there are only 5 facilities but each one takes 10,000 minutes? (it should take 20 days or so to complete each site). Basically we need to test every possible configuration of scheduling and be extremely critical of the outputs to make sure it works! Any other ideas here @KeeganShaw-GIS or @tbarchyn

Reply:
Well, we could create unit tests, which are supposed to test a given method or function in isolation. However, it all comes down to writing the correct tests and that is a challenge. Perhaps step forward here is breaking this up into a few functions to compartmentalize things a little bit more to aid understanding?

Reply:
100%

Reply:
Also, because this is going into the IM3S Branch and not the public branch, we can test after PR.

Reply:
This has been modified alot, is outdated. We will do testing after. The code should be re-reviewed